apps/codeforafrica/src/lib/data/seo.js

```diff
@@ -0,0 +1,67 @@
+import site from "@/codeforafrica/utils/site";
+
+function stringifyDescription(description) {
+  if (!Array.isArray(description)) {
+    return "";
+  }
+  return description.reduce((result, item) => {
+    if (item.text) {
+      // eslint-disable-next-line no-param-reassign
+      result += item.text;
+    }
+
+    if (Array.isArray(item.children)) {
+      // eslint-disable-next-line no-param-reassign
+      result += stringifyDescription(item.children);
+    }
+    return result;
+  }, "");
+}
+
+export default function getPageSeoFromMeta(page, settings) {
+  const { title: pageTitle, meta: pageMeta } = page;
+  const {
+    title: metaTitle,
+    description: metaDescription,
+    image = {},
+  } = pageMeta;
+  const { title: siteTitle, description: siteDescription } = settings;
+  const title =
+    metaTitle ||
+    pageTitle ||
+    siteTitle ||
+    process.env.NEXT_PUBLIC_APP_NAME ||
+    null;
+  const description =
+    metaDescription || stringifyDescription(siteDescription) || null;
+  const titleTemplate = siteTitle ? `%s | ${siteTitle}` : null;
+  const defaultTitle = siteTitle || null;
+  const canonical = site.url.replace(/\/+$/, "");
+  const openGraph = {
+    title,
+    description,
+    type: "website",
```

Comment:
Add a story to track these: we may have to use different types for stories, teams, etc.

Reply:
@kilemensi Already there at #584